src/Main/FightSelectorHeader.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router';
+import ReactTooltip from 'react-tooltip';
+import Toggle from 'react-toggle';
+
+import FightSelectionList from './FightSelectionList';
+
+class FightSelectorHeader extends Component {
+  static propTypes = {
+    show: PropTypes.bool.isRequired,
+    callbackSelectors: PropTypes.func.isRequired,
+    selectedFightName: PropTypes.string.isRequired,
+    parser: PropTypes.shape(),
+    report: PropTypes.shape({
+      code: PropTypes.string.isRequired,
+      title: PropTypes.string.isRequired,
+      fights: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        difficulty: PropTypes.number,
+        boss: PropTypes.number.isRequired,
+        start_time: PropTypes.number.isRequired,
+        end_time: PropTypes.number.isRequired,
+        name: PropTypes.string.isRequired,
+        kill: PropTypes.bool,
+      })),
+    }),
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      killsOnly: false,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentWillUnmount() {
+    ReactTooltip.hide();
+  }
+
+  handleClick(event) {
+    this.props.callbackSelectors('Fights', !this.props.show);
+  }
+
+  render() {
+    const { report, selectedFightName, parser, show } = this.props;
+    const { killsOnly } = this.state;
+
+    return (
+      <span>
+        <Link onClick={this.handleClick}>{selectedFightName}</Link>
+        {show && parser && parser.player &&
+        <span className="selectorHeader">
+          <div className="panel">
+            <div className="panel-heading">
+              <div className="row">
+                <div className="col-md-8">
+                  <h2>Select the fight to parse {parser && parser.player ? ' for ' + parser.player.name : ''}</h2>
+                </div>
+                <div className="col-md-4 text-right toggle-control action-buttons">
+                  <Toggle
```

Comment:
This toggle and its state is still duplicated, you could solve this by making another component around FightSelectionList that manages the state for you. I think extracting the entire panel and making the title a property should allow you to do this.

Reply:
I will fix this later